### PR TITLE
OPNET-465: Refactor LB and DNS manifest list for vSphere

### PR DESCRIPTION
**Pull request by mkowalski**
This PR refactors the manifest generation for vSphere so that instead of
crafting the list of manifests inline, it leverages the existing
`getPlatformManifests` function.

### CODE REVIEW

**Comment by sadasu** [comment]
LGTM
This is nice. Makes the generation of vSphere manifests consistent with what is done for other platforms.

File: pkg/operator/bootstrap.go [hunk 1]
```diff
@@ -235,38 +235,20 @@ func appendManifestsByPlatform(manifests []manifest, infra configv1.Infrastructu
 	}
 
 	if infra.Status.PlatformStatus.VSphere != nil {
-		deployInternalLB := true
-		// vSphere allows to use a user managed load balancer by not setting the VIPs in PlatformStatus.
-		// We will maintain backward compatibility by checking if the VIPs are not set, we will not deploy
-		// Keepalived and CoreDNS.
+		// TODO(mko) It is not clear why for user-managed LB and for ELB we want to skip CoreDNS as every
```
Comment (cybertron) [inline pkg/operator/bootstrap.go:238]: I believe there was some related logic implemented for VSphere that overlapped the external loadbalancer feature. I don't remember all of the details off hand, but that's why there is VSphere-specific code here.

**Review by cybertron — COMMENTED** [review]
/lgtm

We should make sure the vsphere jobs pass, but the change looks good.

**Review by yuqi-zhang — APPROVED** [review]
Just to check, does this need any backporting?

**Comment by mkowalski** [comment]
No, I don't want to backport this anywhere

**Pull request merged**